scripts/suite_objects.py

```diff
@@ -1630,26 +1663,36 @@ def add_var_transform(self, var, compat_obj, vert_dim):
         #hdim = find_horizontal_dimension(var.get_dimensions())
         #if compat_obj.has_dim_transforms:
 
-        #
-        # Register any reverse (pre-Scheme) transforms.
-        #
+        # Register any reverse (pre-Scheme) transforms. Also, save local_name used in
+        # transform (used in write stage).
         if (var.get_prop_value('intent') != 'out'):
-            self.__reverse_transforms.append([dummy.get_prop_value('local_name'),
+            lmsg = "Automatic unit conversion from '{}' to '{}' for '{}' before entering '{}'"
+            self.run_env.logger.info(lmsg.format(compat_obj.v2_units,
+                                                 compat_obj.v1_units,
+                                                 compat_obj.v2_stdname,
+                                                 compat_obj.v1_stdname))
+            self.__reverse_transforms.append([local_trans_var.get_prop_value('local_name'),
                                               var.get_prop_value('local_name'),
-                                              rindices, lindices, compat_obj])
-
-        #
+                                              var.get_prop_value('standard_name'),
+                                              rindices, lindices, compat_obj,
+                                              var.get_prop_value('standard_name')])
```

Comment:
Do we need `var.get_prop_value('standard_name')` in this array twice?

Reply:
Nope. Removed.